pkg/dogstatsd/parser.go

```diff
@@ -280,14 +279,12 @@ func parseMetricMessage(message []byte) (*metrics.MetricSample, error) {
 	}
 
 	metricName := string(rawName)
-	if config.Datadog.GetString("statsd_metric_namespace") != "" {
-		metricPrefix := config.Datadog.GetString("statsd_metric_namespace")
-		if !strings.HasSuffix(metricPrefix, ".") {
-			metricPrefix += "."
-			metricName = metricPrefix + metricName
+	if namespace != "" {
+		if !strings.HasSuffix(namespace, ".") {
```

Comment:
This check should go into the server so we only concatenate here. Also why not append when the namespace already ends by a dot ?